src/index/Model/ProductIndexer.php

```diff
@@ -70,25 +70,31 @@ public function executeFull(): void
     /**
      * @inheritDoc
      *
-     * @throws LocalizedException
      */
     public function executeList(array $ids): void
     {
-        // create temporary table to handle changelogs
-        $this->tempTable->create();
-        foreach ($this->dimensionProvider->getIterator() as $dimension) {
+        try {
+            // create temporary table to handle changelogs
+            $this->tempTable->generateTempTableName();
+            $this->tempTable->create();
+            // try block here is nested to ensure that if the table was created, it gets dropped at the end
             try {
-                $this->executeByDimensions($dimension, new \ArrayIterator($ids));
-            } catch (FileSystemException|RuntimeException) {
-                continue;
+                foreach ($this->dimensionProvider->getIterator() as $dimension) {
+                    try {
+                        $this->executeByDimensions($dimension, new \ArrayIterator($ids));
+                    } catch (FileSystemException|RuntimeException) {
+                        continue;
+                    }
+                }
+                $this->insertChangelogRecords->execute();
+            } finally {
+                // we want to ensure that the "temporary" table is always dropped
+                $this->tempTable->drop();
             }
-        }
-        try {
-            $this->insertChangelogRecords->execute();
         } catch (\Exception $e) {
             $this->logger->critical($e->getMessage(), ['exception' => $e]);
         }
```

Comment:
does the 
` finally {
                // we want to ensure that the "temporary" table is always dropped
                $this->tempTable->drop();
            }  ` needs to be in here?

Reply:
I just wanted to be explicit

Reply:
@indunil15 I agree with you, I don't think it matters, where it is is fine, as the `tempTable->drop()` will always be executed either way. But it just feels like `$this->tempTable->drop()` should be in a new `finally` right at the bottom here :sweat_smile:

Reply:
I realised it bit later :laughing: 
If line 78,79 goes wrong there will be no temp table and no needs to delete it :+1: